src/device/rsp/rsp_core.c

```diff
@@ -306,27 +304,45 @@ void do_SP_Task(struct rsp_core* sp)
         rsp.doRspCycles(0xffffffff);
         timed_section_end(TIMED_SECTION_AUDIO);
         sp->regs2[SP_PC_REG] |= save_pc;
+
+        sp_delay_time = 4000;
     }
     else
     {
         sp->regs2[SP_PC_REG] &= 0xfff;
         rsp.doRspCycles(0xffffffff);
         sp->regs2[SP_PC_REG] |= save_pc;
+
+        sp_delay_time = 0;
     }
 
     sp->rsp_task_locked = 0;
     if ((sp->regs[SP_STATUS_REG] & (SP_STATUS_HALT | SP_STATUS_BROKE)) == 0)
```

Comment:
I tested WDC, Stunt Racer and Gauntlet. They all ran, save states worked sometimes, but still not reliably. Also, there was some graphical issues I saw in Stunt Racer, not present before this change (see the blue box under "Soda Mtn"). I noticed you changed the SP_LOCKED time to 2000. I assume fiddling with that number might solve the issue, I tried making it 1000 but then the games no longer got past the title screen.

![stunt_racer_64-001](https://user-images.githubusercontent.com/848146/29622417-a3acdc4a-87e0-11e7-988d-55f5ad505012.png)